pandas/tests/indexes/common.py

```diff
@@ -34,6 +34,12 @@ def verify_pickle(self, index):
         unpickled = tm.round_trip_pickle(index)
         assert index.equals(unpickled)
 
+    def _fix_tz(self, new_index, orig_index):
+        if hasattr(orig_index, 'tz'):
+            assert new_index.tz is None
+            new_index = new_index.tz_localize('UTC').tz_convert(orig_index.tz)
```

Comment:
create a new class DatetimeTZ in ``test_datetimelike.py`` which also inherits ``Datetimelike``. Then just change its ``create_index``

Reply:
To clarify, this commit already introduces a new `Datetimelike`-derived class in `test_datetimelikely.py` with `create_index()` overriden to return a tz-aware `DatetimeIndex`: [test_datetimelike.py#L159](https://github.com/azjps/pandas/blob/b32290ad0e8cbde4e87a0569223d4e6690a986ed/pandas/tests/indexes/datetimes/test_datetimelike.py#L159). This `_fix_tz` hack here is not to create the underlying index to be tested, but to "fix" various transformations of the index which don't preserve TZ, to get them to compare correctly back to the output of `create_index()`. But like you mentioned in the comment below, we can move more of this to class members.